### ISSUE TITLE
On Windows, subprocess.run gets stuck indefinitely

### ISSUE THREAD

**Issue by 89f4240f-251b-48e9-858b-0ad5ad4ede03** [python/cpython#88693]
BPO | [44527](https://bugs.python.org/issue44527)
--- | :---
Nosy | @pfmoore, @ericvsmith, @tjguk, @zware, @zooba
Superseder | <li>bpo-43346: subprocess.run() sometimes ignores timeout in Windows</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2021-06-28.18:32:34.840>
created_at = <Date 2021-06-28.17:29:19.348>
labels = ['type-bug', 'library', '3.9', 'OS-windows']
title = 'On Windows, subprocess.run gets stuck indefinitely'
updated_at = <Date 2021-06-28.18:32:34.840>
user = 'https://bugs.python.org/bugalebugale'
```

bugs.python.org fields:
```python
activity = <Date 2021-06-28.18:32:34.840>
actor = 'eryksun'
assignee = 'none'
closed = True
closed_date = <Date 2021-06-28.18:32:34.840>
closer = 'eryksun'
components = ['Library (Lib)', 'Windows']
creation = <Date 2021-06-28.17:29:19.348>
creator = 'bugale bugale'
dependencies = []
files = []
hgrepos = []
issue_num = 44527
keywords = []
message_count = 1.0
messages = ['396653']
nosy_count = 6.0
nosy_names = ['paul.moore', 'eric.smith', 'tim.golden', 'zach.ware', 'steve.dower', 'bugale bugale']
pr_nums = []
priority = 'normal'
resolution = 'duplicate'
stage = 'resolved'
status = 'closed'
superseder = '43346'
type = 'behavior'
url = 'https://bugs.python.org/issue44527'
versions = ['Python 3.9']
```

</p></details>

**Comment by yilmi** [comment]
While we couldn't find an explanation, we hope this may help with the resolution of this issue.

One of the workarounds identified when facing this issue has been to set `stdin=subprocess.DEVNULL` when calling subprocess.run() or subprocess.Popen(). This seems to solve the hang and the process returns properly. The issue and workaround were encountered with Python 3.13.3 and seems to indicate a problem with the stdin instead of stdout.

**Comment by zware** [comment]
This is a closed duplicate of #87512, please comment there instead.
